tests/tasks/test_perturbation_expression_prediction.py

```diff
@@ -9,40 +9,93 @@
-    PerturbationExpressionPredictionTaskInput,
     build_task_input_from_predictions,
 )
 from czbenchmarks.datasets.types import Organism
 from czbenchmarks.datasets.single_cell_perturbation import (
     SingleCellPerturbationDataset,
 )
-from czbenchmarks.metrics.types import MetricResult, MetricType
+from czbenchmarks.metrics.types import MetricType
 from tests.utils import create_dummy_perturbation_anndata, create_dummy_anndata
 
 
-# TODO this needs a significant refactor
+# Test-specific fixtures
+
+
+@pytest.fixture
+def wilcoxon_test_data():
+    """Create deterministic test data for Wilcoxon testing."""
+    gene_names = ["G0", "G1", "G2", "G3"]
+    true_lfc_gene_A = np.array([1.0, 0.5, -0.5, -1.0])
+    true_lfc_gene_B = np.array([2.0, 1.0, -1.0, -2.0])
+
+    # Create DE results for both conditions
+    conditions_data = [
+        ("gene_A", true_lfc_gene_A, "ENSG_A"),
+        ("gene_B", true_lfc_gene_B, "ENSG_B"),
+    ]
+
+    de_results = pd.concat(
+        [
+            pd.DataFrame(
+                {
+                    "logfoldchange": lfc_values,
+                    "target_gene": [condition] * len(gene_names),
+                    "names": gene_names,
+                    "pval": [0.001] * len(gene_names),
+                    "pval_adj": [0.001] * len(gene_names),
+                    "condition": [condition] * len(gene_names),
+                    "condition_ensembl_id": [ensembl_id] * len(gene_names),
+                }
+            )
+            for condition, lfc_values, ensembl_id in conditions_data
+        ],
+        ignore_index=True,
+    )
+
+    return {
+        "de_results": de_results,
+        "true_lfc_gene_A": true_lfc_gene_A,
+        "true_lfc_gene_B": true_lfc_gene_B,
+        "gene_names": gene_names,
+    }
+
+
+@pytest.fixture
+def simple_toy_data():
+    """Create simple toy dataset for pair mapping tests."""
+    X = np.array([[1.0, 2.0], [2.0, 1.0], [3.0, 4.0], [5.0, 3.0]])
+    obs = pd.DataFrame(
+        {"condition": ["ctrl", "ctrl", "pertA", "pertA"]},
+        index=["NT1", "NT2", "T1", "T2"],
+    )
+    adata = ad.AnnData(X=X, obs=obs, var=pd.DataFrame(index=["g1", "g2"]))
+
+    de_results = pd.DataFrame(
+        {
+            "condition": ["pertA", "pertA"],
+            "gene_id": ["g1", "g2"],
+            "logfoldchange": [1.0, 2.0],
+            "pval_adj": [1e-6, 1e-4],
+        }
+    )
+
+    return {"adata": adata, "de_results": de_results, "predictions": np.log1p(X)}
 
 
-def test_perturbation_task():
+def test_perturbation_task_executes_without_errors(
+    adata_uns_setup, assert_metric_results
+):
     """Test that PerturbationExpressionPredictionTask executes without errors."""
-    # Create dummy perturbation data
-    perturbation_data: dict = create_dummy_perturbation_anndata(
-        n_cells=500,
-        n_genes=200,
-        organism=Organism.HUMAN,
-        condition_column="condition",
+    perturbation_data = create_dummy_perturbation_anndata(
```

Comment:
This is an outdated function that generates perturbation data in a format that is no longer supported. It shouldn't be used to generate data given that there are more relevant methods. Instead it should be removed.